### PR TITLE
Add integration test and optimize storage

### PR DESCRIPTION
**Pull request by iboss-ptk**
- add integration tests
- remove go test and cw-multitest's tests since it's got all covered 
- clean up names and redundant code
- remove allowances from storage when it reaches 0 (mint, burn)
- remove status if equal false (freeze, blacklister, blacklistee)
- segregate config into it's own item to avoid meaningless transient state (namely `config.denom` with empty string)

### CODE REVIEW

File: contracts/tokenfactory-issuer/src/contract.rs [hunk 1]
```diff
@@ -64,13 +57,7 @@ pub fn instantiate(
                 ))
         }
         InstantiateMsg::ExistingToken { denom } => {
-            let config = Config {
-                owner: info.sender.clone(),
-                is_frozen: false,
```
Comment (sunnya97) [inline contracts/tokenfactory-issuer/src/contract.rs:69]: is_frozen will default to false if not explicitly set?

File: contracts/tokenfactory-issuer/src/execute.rs [hunk 2]
```diff
@@ -21,20 +21,26 @@ pub fn mint(
...
-            .unwrap_or_else(Uint128::zero)
-            .checked_sub(amount)
-            .map_err(StdError::overflow)
-    })?;
+    // if minter allowance goes 0, remove from storage
+    let allowance = MINTER_ALLOWANCES.may_load(deps.storage, &info.sender)?;
+    let updated_allowance = allowance
+        .unwrap_or_else(Uint128::zero)
+        .checked_sub(amount)
+        .map_err(StdError::overflow)?;
```
Comment (sunnya97) [inline contracts/tokenfactory-issuer/src/execute.rs:34]: We should probably make an explicit contract error type for this

Comment (iboss-ptk) [reply contracts/tokenfactory-issuer/src/execute.rs:34]: That make sense, maybe `NotEnoughAllowance`

Comment (iboss-ptk) [reply contracts/tokenfactory-issuer/src/execute.rs]: @sunnya97 updated, does that looks good?

**Pull request merged**